woodwork/logical_types.py

```diff
@@ -306,6 +310,16 @@ class EmailAddress(LogicalType):
 
     primary_dtype = "string"
 
+    def validate(self, series, return_indices=False):
+        regex = r"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b"
+        valid = series.str.fullmatch(regex).astype("boolean")
+
+        if return_indices:
+            return valid.rename(series.name)
+
+        elif (~valid).any():
+            raise ValueError("email address not understood")
```

Comment:
What do you think about update this message to be something like "Series `{series.name}` contains invalid email addresses"?

Reply:
Maybe we can note that the email address regex can be changed in the config if needed?

Reply:
updated